src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.TargetFrameworkInference.targets

```diff
@@ -96,6 +96,38 @@ Copyright (c) .NET Foundation. All rights reserved.
   <PropertyGroup Condition="'$(TargetFrameworkVersion)' == ''">
     <TargetFrameworkVersion >v0.0</TargetFrameworkVersion>
   </PropertyGroup>
+  
+  <!--
+    Trigger an error if targeting a higher version of .NET Core or .NET Standard than is supported by the current SDK.
+  -->
+
+  <PropertyGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' And '$(NETCoreAppMaximumVersionExclusive)' == ''">
+    <NETCoreAppMaximumVersionExclusive>1.2</NETCoreAppMaximumVersionExclusive>
+    <NETCoreAppMaximumVersionHumanReadable>1.1</NETCoreAppMaximumVersionHumanReadable>
+  </PropertyGroup>
+    
+  <Target Name="_CheckForUnsupportedNETCoreVersion" BeforeTargets="_CheckForInvalidConfigurationAndPlatform;Restore;CollectPackageReferences"
+          Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' And '$(NETCoreAppMaximumVersionExclusive)' != ''">
+
+    <NETSdkError Condition="'$(_TargetFrameworkVersionWithoutV)' >= '$(NETCoreAppMaximumVersionExclusive)'"
+                 ResourceName="UnsupportedTargetFrameworkVersion"
+                 FormatArguments=".NET Core;$(_TargetFrameworkVersionWithoutV);$(NETCoreAppMaximumVersionHumanReadable)"
+      />
+  </Target>
+
+  <PropertyGroup  Condition="'$(TargetFrameworkIdentifier)' == '.NETStandard' And '$(NETStandardMaximumVersionExclusive)' == ''">
+    <NETStandardMaximumVersionExclusive>1.7</NETStandardMaximumVersionExclusive>
+    <NETStandardMaximumVersionHumanReadable>1.6</NETStandardMaximumVersionHumanReadable>
```

Comment:
@livarcocc @nguerrera here is the 2 string version. the naming should be obvious enough